### PR TITLE
TRI-58/Get all tickets with filters

### PR DESCRIPTION
**Pull request by Juan-Bianchi**
# Pull Request

## Description
This endpoint retrieves an array of IssueViewDTO objects, representing the general view of issues for a Project Manager.

## Ticket Link
[TRI-58](https://linear.app/tricker-sirius/issue/TRI-58/get-all-tickets-with-filters)

## Steps to Reproduce
1. Run `npx prisma generate`
2. Run `docker compose up`
3. Check Swagger documentation for this [endpoint](http://localhost:8080/api-docs/#/Issue/post_api_issue_pm__userId__project__projectId_)
4. Feel free to play with the endpoint request.

## Images/Screenshots (if necessary)
Please add any images or screenshots that may help to explain the issue or the changes made.

## Additional Notes (if any)
[Add any other context about the pull request here.]

### CODE REVIEW

File: src/domains/issue/controller/issue.controller.ts [hunk 1]
```diff
@@ -26,6 +30,15 @@ issueRouter.post('/dev/:userId/project/:projectId', validateRequest(UserProjectP
   return res.status(HttpStatus.OK).json(issues);
 });
 
+issueRouter.post('/pm/:userId/project/:projectId', validateRequest(UserProjectParamsDTO, 'params'), validateRequest(PMOptionalIssueFiltersDTO, 'body'), async (req: Request<UserProjectParamsDTO, any, PMOptionalIssueFiltersDTO>, res: Response) => {
```
Comment (luccar31) [inline src/domains/issue/controller/issue.controller.ts]: Same as before, filtering should be provided by query params and request should be a HTTP GET. Just a comment. POST is okay.

File: src/domains/event/dto/index.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export class IssueChangeLogDTO {
 
 export class BlockerStatusModificationDTO {
```
Comment (luccar31) [inline src/domains/event/dto/index.ts:88]: If it happens that you find something not documented, please, try to add it.

File: src/domains/issue/dto/index.ts [hunk 3]
```diff
@@ -349,6 +350,12 @@ export class IssueViewDTO {
     this.title = issueView.title;
     this.priority = issueView.priority;
     this.storyPoints = issueView.storyPoints;
+    this.isBlocked = issueView.isBlocked;
     this.labels = issueView.labels;
   }
 }
+
+export interface UserProject {
```
Comment (luccar31) [inline src/domains/issue/dto/index.ts:361]: Please, add documentation.

File: src/domains/userProjectRole/repository/userProjectRole.repository.impl.ts [hunk 4]
```diff
@@ -18,4 +19,15 @@ export class UserProjectRoleRepositoryImpl implements UserProjectRoleRepository
 
     return new UserProjectRoleDTO(userProjectRole);
   }
+
+  async getByProjectIdAndUserId(input: UserProject): Promise<UserProjectRoleDTO | null> {
```
Comment (luccar31) [inline src/domains/userProjectRole/repository/userProjectRole.repository.impl.ts:28]: Please, add documentation.

File: src/domains/issue/controller/issue.controller.ts [hunk 5]
```diff
@@ -15,7 +17,9 @@ const issueRepo: IssueRepository = new IssueRepositoryImpl(db);
 const eventRepo: EventRepository = new EventRepositoryImpl(db);
 const userRepo: UserRepository = new UserRepositoryImpl(db);
 const projectRepo: ProjectRepository = new ProjectRepositoryImpl(db);
-const issueService: IssueService = new IssueServiceImpl(issueRepo, eventRepo, userRepo, projectRepo);
+const userProjectRoleRepo: UserProjectRoleRepository = new UserProjectRoleRepositoryImpl(db);
+const roleRepo: RoleRepository = new RoleRepositoryImpl(db);
+const issueService: IssueService = new IssueServiceImpl(issueRepo, eventRepo, userRepo, projectRepo, userProjectRoleRepo, roleRepo);
 
 issueRouter.post('/dev/:userId/project/:projectId', validateRequest(UserProjectParamsDTO, 'params'), validateRequest(DevOptionalIssueFiltersDTO, 'body'), async (req: Request<UserProjectParamsDTO, any, DevOptionalIssueFiltersDTO>, res: Response) => {
```
Comment (luccar31) [inline src/domains/issue/controller/issue.controller.ts]: As a suggestion, I think user roles (authorization) must be checked before the flow reaches the service. You can do it as a middleware and return a 403 code.

**Review by luccar31 — CHANGES_REQUESTED** [review]

File: src/domains/issue/controller/issue.controller.ts (hunk 1 above)
Comment (Juan-Bianchi) [reply src/domains/issue/controller/issue.controller.ts]: Yes, I had already finished this endpoint when you made last recommendation. For future endpoints, we will user the library.

File: src/domains/event/dto/index.ts (hunk 2 above)
Comment (Juan-Bianchi) [reply src/domains/event/dto/index.ts:88]: done

File: src/domains/issue/dto/index.ts (hunk 3 above)
Comment (Juan-Bianchi) [reply src/domains/issue/dto/index.ts:361]: done

File: src/domains/userProjectRole/repository/userProjectRole.repository.impl.ts (hunk 4 above)
Comment (Juan-Bianchi) [reply src/domains/userProjectRole/repository/userProjectRole.repository.impl.ts:28]: done

File: src/domains/issue/controller/issue.controller.ts (hunk 5 above)
Comment (Juan-Bianchi) [reply src/domains/issue/controller/issue.controller.ts]: Middleware created

**Review by luccar31 — APPROVED** [review]

**Pull request merged**